pilot/pkg/networking/plugin/plugin.go

```diff
@@ -46,15 +50,26 @@ const (
 	Mixer = "mixer"
 )
 
-// ModelProtocolToListenerProtocol converts from a config.Instance to its corresponding plugin.ListenerProtocol
-func ModelProtocolToListenerProtocol(p protocol.Instance) ListenerProtocol {
+// ModelProtocolToListenerProtocol converts from a config.Protocol to its corresponding plugin.ListenerProtocol
+func ModelProtocolToListenerProtocol(node *model.Proxy, p protocol.Instance) ListenerProtocol {
+	// If protocol sniffing is not enabled, the default value is TCP
+	if !util.IsProtocolSniffingEnabled(node) && p == protocol.Unsupported {
+		p = protocol.TCP
+	}
+
 	switch p {
 	case protocol.HTTP, protocol.HTTP2, protocol.GRPC, protocol.GRPCWeb:
 		return ListenerProtocolHTTP
 	case protocol.TCP, protocol.HTTPS, protocol.TLS,
 		protocol.Mongo, protocol.Redis, protocol.MySQL:
 		return ListenerProtocolTCP
+	case protocol.UDP:
+		return ListenerProtocolUnknown
 	default:
+		if util.IsProtocolSniffingEnabled(node) {
+			return ListenerProtocolAuto
+		}
+
```

Comment:
it should be TCP otherwise, right?

Reply:
Yes. I set it to TCP at the beginning of this function.